### ISSUE TITLE
USWDS-Site - Feature: Add Complex Table Guidance

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds-site#2263]
### Is your feature request related to a problem? Please describe.

The tables guidance is great: https://designsystem.digital.gov/components/table/ but it lacks guidance on how to handle coding complex tables (and the differences between simple and complex tables).

[WCAG has guidance ](https://www.w3.org/WAI/tutorials/tables/) and tutorials.

### Describe the solution you'd like

- Explain the differences between one, two, and multi-level headers and include an example with a properly coded complex table.
- Expand on existing guidance for accessible simple vs. complex tables. 
- Include guidance on using tables for data and not design (and the difference between them)
- Include recommendation to not merge data cells
-  Include guidance on captions and summaries. In this case, not a citation caption but instead a descriptive caption. Some tables are very complex and need summaries which give the overall gist of a table with highlights to help clarify complex data.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.
